cmd/web-handlers.go

```diff
@@ -546,10 +546,15 @@ func (web *webAPIHandlers) Download(w http.ResponseWriter, r *http.Request) {
 	objectLock.RLock()
 	defer objectLock.RUnlock()
 
-	if err := objectAPI.GetObject(bucket, object, 0, -1, w); err != nil {
+	info, err := objectAPI.GetObjectInfo(bucket, object)
```

Comment:
One problem here is that this would increase an API call to the backend server. We should avoid that handle this appropriately at the client SDK used by the s3 backend.